src/test/java/org/opentripplanner/standalone/config/RouterConfigTest.java

```diff
@@ -0,0 +1,69 @@
+package org.opentripplanner.standalone.config;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.opentripplanner.standalone.config.RouterConfig.DEFAULT;
+
+import java.time.OffsetDateTime;
+import java.util.TimeZone;
+import org.junit.jupiter.api.Test;
+import org.opentripplanner.model.Agency;
+import org.opentripplanner.model.FeedScopedId;
+import org.opentripplanner.model.calendar.CalendarServiceData;
+import org.opentripplanner.routing.api.request.RoutingRequest;
+import org.opentripplanner.routing.graph.Graph;
+
+class RouterConfigTest {
+
+    @Test
+    void middleOfDay() {
+        var req = getReq("2022-01-25T13:14:20+01:00");
+        assertEquals(0, DEFAULT.additionalSearchDaysInFuture(req));
+        assertEquals(0, DEFAULT.additionalSearchDaysInPast(req));
+
+        req.arriveBy = true;
+        assertEquals(0, DEFAULT.additionalSearchDaysInFuture(req));
+        assertEquals(0, DEFAULT.additionalSearchDaysInPast(req));
+    }
+
+    @Test
+    void closeToMidnight() {
+        var req = getReq("2022-01-25T23:14:20+01:00");
+
+        assertEquals(0, DEFAULT.additionalSearchDaysInPast(req));
+        assertEquals(1, DEFAULT.additionalSearchDaysInFuture(req));
+
+        req.arriveBy = true;
+        assertEquals(0, DEFAULT.additionalSearchDaysInPast(req));
+        assertEquals(0, DEFAULT.additionalSearchDaysInFuture(req));
+    }
+
+    @Test
+    void shortlyAfterMidnight() {
+        var req = getReq("2022-01-25T00:15:25+01:00");
+
+        assertEquals(0, DEFAULT.additionalSearchDaysInPast(req));
+        assertEquals(0, DEFAULT.additionalSearchDaysInFuture(req));
+
+        req.arriveBy = true;
+        assertEquals(1, DEFAULT.additionalSearchDaysInPast(req));
+        assertEquals(0, DEFAULT.additionalSearchDaysInFuture(req));
+    }
+
+    private RoutingRequest getReq(String time) {
+        var feedId = new FeedScopedId("1", "1");
+
+        var cal = new CalendarServiceData();
+        cal.putTimeZoneForAgencyId(feedId, TimeZone.getTimeZone("Europe/Berlin"));
+
+        var graph = new Graph();
```

Comment:
Instantiating a graph to test his kind of simple logic seem wrong - it indicate that there is some coupling that is not right.

Reply:
I totally agree - it's horrible. But converting the search time, which is in seconds since the epoch, into a `ZonedDateTime` or `LocalDateTime` requires you to specify a timezone, which currently resides in the Graph instance - for better or worse.

How do you suggest we go around this?

Reply:
You should pass in the `searchStartTime` calculated int the `RoutingWorker`, not calculating it again. See my comment on RoutingRequest.java.

Reply:
Yeah, I think I got it now and have adjusted the PR accordingly.